src/main/java/se/bjurr/violations/lib/example/CopyOfMyClass.java

```diff
@@ -0,0 +1,19 @@
+package se.bjurr.violations.lib.example;
+
+public class CopyOfMyClass {
+ public static String CONstant = "yes";
+
+ void npe(String a, String b) {
+  if (a == null) {
+   System.out.println();
+  } else {
+
+  }
+  a.length();
+ }
+
+ @Override
+ public boolean equals(Object obj) {
+  return true;
```

Comment:
**Reporter**: Findbugs
**Rule**: HE_EQUALS_USE_HASHCODE
**Severity**: WARN
**File**: src/main/java/se/bjurr/violations/lib/example/CopyOfMyClass.java L17
**Source**: se.bjurr.violations.lib.example.CopyOfMyClass

Class defines equals() and uses Object.hashCode()


  <p> This class overrides <code>equals(Object)</code>, but does not
  override <code>hashCode()</code>, and inherits the implementation of
  <code>hashCode()</code> from <code>java.lang.Object</code> (which returns
  the identity hash code, an arbitrary value assigned to the object
  by the VM).&nbsp; Therefore, the class is very likely to violate the
  invariant that equal objects must have equal hashcodes.</p>

<p>If you don't think instances of this class will ever be inserted into a HashMap/HashTable,
the recommended <code>hashCode</code> implementation to use is:</p>
<pre>public int hashCode() {
  assert false : "hashCode not designed";
  return 42; // any arbitrary constant will do
  }</pre>


*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<-1804119430>*